### PR TITLE
JENKINS-47254 : support bitbucket tags (another attempt)

### PR DESCRIPTION
**Pull request by mindfulmonk**
Picking up the work @eddi-weiss did in https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/113

### CODE REVIEW

**Comment by mindfulmonk** [comment]
I will try, if you can provide more info or a simpler example of writing a trait test that would be awesome ;)

**Comment by amuniz** [comment]
Maybe [this one is simpler](https://github.com/jenkinsci/bitbucket-branch-source-plugin/blob/dd92b6ab1c7865350df8f5745138eed68e45a35c/src/test/java/com/cloudbees/jenkins/plugins/bitbucket/PublicRepoPullRequestFilterTraitTest.java#L17)? You can find a lot of examples under [src/test folder](https://github.com/jenkinsci/bitbucket-branch-source-plugin/tree/dd92b6ab1c7865350df8f5745138eed68e45a35c/src/test/java/com/cloudbees/jenkins/plugins/bitbucket).

**Comment by mindfulmonk** [comment]
@amuniz test added, please let us know if there is anything else to add ;)

**Comment by casz** [comment]
@mindfulmonk please fix the star imports to explicit 🚔

**Comment by mindfulmonk** [comment]
How about now?

File: src/test/java/com/cloudbees/jenkins/plugins/bitbucket/TagDiscoveryTraitTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        assumeThat(ctx.authorities(), not((Matcher) hasItem(instanceOf(TagSCMHeadAuthority.class))));
+
+        TagDiscoveryTrait instance = new TagDiscoveryTrait();
+        instance.decorateContext(ctx);
+        assertThat(ctx.wantTags(), is(true));
+        assertThat(ctx.prefilters(), is(Collections.<SCMHeadPrefilter>emptyList()));
+        assertThat(ctx.filters(), is(Collections.<SCMHeadFilter>emptyList()));
+        assertThat(ctx.authorities(), (Matcher) hasItem(instanceOf(TagSCMHeadAuthority.class)));
+    }
+}
```
Comment (casz) [inline src/test/java/com/cloudbees/jenkins/plugins/bitbucket/TagDiscoveryTraitTest.java]: missing newline 🚔

**Comment by mindfulmonk** [comment]
@casz anything else? :)

**Comment by aheritier** [comment]
@amuniz could you have a look at the tests please ?

**Pull request merged**